src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -69,6 +69,12 @@ protected Index(int indexId, IndexType type, TIndexDefinition definition)
 
     public abstract class Index : IDocumentTombstoneAware, IDisposable
     {
+        public bool SideBySideIndex;
+
+        public long? MinimumEtagBeforeReplace { get; set; }
+
+        public string RealName { get; set; }
```

Comment:
Wrap all of those in ReplacementInformation. It is very confusing to have all of those available on each index instance , especially the 'RealName' which will be null in 99% of the time.